krm-functions/lib/kptrl/kptrl.go

```diff
@@ -0,0 +1,133 @@
+/*
+ Copyright 2023 Nephio.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package kptrl
+
+import (
+	"sync"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+type ResourceList interface {
```

Comment:
Ignoring the concurrency controls (which I don't think are needed), this seems to mostly wrap fn.ResourceList and add a couple methods. 

I think it would be simpler and more idiomatic to instead do:

```
type ResourceList struct {
  fn.ResourceList
  ...add new methods here...
}
```

And just let folks access the underlying fields directly, rather than having Getters that just return the fiedl from the wrapped object.

That would eliminate these methods, because you would just access the underlying fields instead:
- AddResult
- GetResults
- GetObjects

So basically, what we would have in the end is a library that adds the GetObject, SetObject, DeleteObject convenience methods.

Reply:
fn.ResourceList is not an interface. What we could do is add the methods the resourceList offers and these to the interface?

Reply:
Eventually all of this should end up in the kpt fn sdk in my view.

Reply:
Sorry, right, you can't add new methods there. My point would have been to use a struct and just allow access to the underlying fn.ResourceList and then only implement the new functions that do extra stuff on top of that.

I still don't think the mutex is needed based on how these objects are actually used, and while it doesn't add significant runtime overhead, it will raise questions (like it has for me).

Beyond that, most of the methods are single line - except GetObject and DeleteObject, I think. So, apart from those two, does it make sense to add another layer folks have to learn on top of the SDK?

Reply:
If thread safety is not needed, agree to the point that most of the methods can be directed  to the underlying rl object.

Reply:
ok I removed the thread safety and add a struct that holds the rl as John suggested. Let me know if this is ok